### DISCUSSION TITLE
lint of source code

### DISCUSSION THREAD

**Discussion by massimosala** [koka-lang/koka#300] [General]
Is it possible to run a "lint" to perform strong checks on the source code?
[lint](https://en.wikipedia.org/wiki/Lint_(software))

I believe running lint should be mandatory in the development cycle of robust applications.

Sincerely, Sala
massimo.sala.71@gmail.com

**Comment by timjs** [comment]
Hi Sala,

What kind of things would you like a linter to check for Koka, that the compiler does not do yet? As Koka's type system is so strong, the compiler already does a lot of checks (and more!) that external linters do for other programming languages...

**Reply by massimosala** [reply]
Like unitialized variables; inconsistent usage of variables of different types (like signed / unsigned); inconsistent usage of pointers to memory; etc...
Ok, I get it.

Does the compiler does other checks? like check for writes to unitialized memory?

**Reply by timjs** [reply]
The compiler does all the checks you mention already.
